xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -178,7 +186,8 @@ protected void handleLdsResponse(String versionInfo, List<Any> resources, String
         if (listener.hasApiListener()) {
           ldsUpdate = processClientSideListener(listener, enableFaultInjection && isResourceV3);
         } else {
-          ldsUpdate = processServerSideListener(listener);
+          // TODO(chengyuanzhang): change env var for supporting HttpFilter.
```

Comment:
Is this comment applicable only to the server  side or client side too i.e. line 187?

Reply:
This refers to the `enableFaultInjection` env var. Currently, we do not have a separate `enableHttpFilter` env var. Instead, setting `enableFaultInjection` to `true` enables both HttpFilter and fault inject. It is fine by now as fault inject is the only use case of HttpFilter. But we'd really want both `enableHttpFilter` and `enableFaultInjection`.

Reply:
As I mentioned elsewhere there is no fault-injection support on the server side so using `enableFaultInjection` on the server side is confusing/wrong?

Reply:
You are right, as I commented in the previous response. The existing `enableFaultInjection` is actually a union of `enableFaultInjection` and `enableHttpFilter` (as the fault inject is currently the only use case of HttpFilter). We'd really  want it be split into `enableFaultInjection` and `enableHttpFilter` separately. Should I do that in this PR or next? Let me see how the code will spread. Also, I need to look up if the interop test has `enableHttpFilter`.

Reply:
After offline discussion, we are not going to introduce another env var for gating HttpFilters. The FI env var will be gone soon. So at this time, we will just require setting the FI env var for using HttpFilters no the server side.